UN-OCHA/rwint9-site
# [RW-739] Allow changing attachment download file name.

Refs: RW-739

This updates the report attachment file widget to allow changing the download file name (the filename part of the download URL, the downloaded file stays the same).

To do so, a "view file name details" is added below the file link in the widget to display the uploaded file name (to ease identify what file was last uploaded for the field item) and a field to change the download file name:

<img width="908" alt="Screen Shot 2023-01-27 at 18 43 00" src="https://user-images.githubusercontent.com/696348/215056495-8bfcd200-fecb-42ee-8e35-8b869c758c60.png">

The filename is both validated in the frontend and in the backend.

### Tests

1. Checkout this branch
2. Clear the cache
3. Log in as an Editor
4. Edit or create a report, attach some files and save as published and take notes of the file URLs in the report page
5. Edit the report, click "view file name details" for an attachment, change the download file name and save
6. Check that the `filename` part of the URL for this attachment changed to the new file name.
7. Check the revisions at the bottom of the page to confirm the file name change.

To go further, edit the report, attach a new file, change the file name (do not save yet), replace a file, re-order them etc. and after saving ensure that the attachments on the report page match the attachments in the edit form (order, download file names etc.).
